### PR TITLE
S3 answer question backend

### PR DESCRIPTION
**Pull request by NunoDasNeves**

### CODE REVIEW

**Comment by NunoDasNeves** [comment]
@DarkPurple141 senpai show me how to write a test for the controller/model or something

File: backend/src/models/answer.js [hunk 1]
```diff
@@ -10,3 +10,16 @@ exports.getAnswers = function (questionID, pageNumber) {
...
+
+exports.postAnswer = function (questionID, {body}) {
+    return new Promise((resolve, reject) => {
+        const columns = ['userID', 'questionID', 'body']
+        const placeholders = columns.map(_ => '?').join()
+        const query = `INSERT INTO answer (${columns}) VALUES (${placeholders})`
+        db.run(
+            query,
+            [0, questionID, body], // TODO user id is a placeholder obviously, but it can't be null so...
+            function (err) { err ? reject(err) : resolve(err) }
```
Comment (PapaPwn) [inline backend/src/models/answer.js]: Resolving err?

Comment (NunoDasNeves) [reply backend/src/models/answer.js]: Actually, I want it to resolve getAnswers so it returns the updated list of answers...
This doesn't seem to work though `resolve(exports.getAnswers(questionID, 1))

Comment (NunoDasNeves) [reply backend/src/models/answer.js]: hmm yeah idk what's going on; i just get `TypeError: Cannot read property 'then' of undefined`

**Comment by DarkPurple141** [comment]
I can’t seem to see these comments. Will have a good look later though.

File: backend/src/models/answer.js [hunk 2]
```diff
@@ -10,3 +10,16 @@ exports.getAnswers = function (questionID, pageNumber) {
...
+
+exports.postAnswer = function (questionID, {body}) {
+    new Promise((resolve, reject) => {
+        const columns = ['userID', 'questionID', 'body']
+        const placeholders = columns.map(_ => '?').join()
+        const query = `INSERT INTO answer (${columns}) VALUES (${placeholders})`
+        db.run(
+            query,
+            [0, questionID, body], // TODO user id is a placeholder obviously, but it can't be null so...
+            function (err) { err ? reject(err) : resolve(exports.getAnswers(questionID, 1)) }
```
Comment (NunoDasNeves) [inline backend/src/models/answer.js]: @DarkPurple141 Basically I don't get what's going on here (this code doesn't work).
I wanna return getAnswers after adding the new answer to the db

**Comment by DarkPurple141** [comment]
Found the comments. K having a look

File: backend/src/models/answer.js (hunk 2 above)
Comment (DarkPurple141) [reply backend/src/models/answer.js]: Fixed with a missing return XD.

File: backend/src/controllers/question.js [hunk 3]
```diff
@@ -13,3 +13,10 @@ exports.getQuestionAnswers = function ({ params, query }, res) {
     answerModel.getAnswers(params.id, query.p)
         .then(data => res.json(data))
 }
+
+/* POST new answer. */
+exports.postAnswer = function ({ params, body }, res) {
+    answerModel.postAnswer(params.id, body)
+        .then(data => res.json(data))
+        .catch(console.warn)
```
Comment (DarkPurple141) [inline backend/src/controllers/question.js]: Not saying in this PR, but eventually just doing console.warns for real errors may not be the best idea.

File: frontend/src/store/questions/constants.js [hunk 4]
```diff
@@ -1,23 +1,27 @@
...
+  [ACTIONS.GET_ANSWERS]: getAnswers,
+  [ACTIONS.POST_ANSWER]: postAnswer
 }
 
 export const COMMITS = {
   [ACTIONS.GET_QUESTIONS]: 'REFRESH_FEED',
   [ACTIONS.GET_QUESTION]: 'FOCUS_QUESTION',
-  [ACTIONS.GET_ANSWERS]: 'FOCUS_ANSWERS'
+  [ACTIONS.GET_ANSWERS]: 'FOCUS_ANSWERS',
+  [ACTIONS.POST_ANSWER]: 'FOCUS_ANSWERS'
```
Comment (DarkPurple141) [inline frontend/src/store/questions/constants.js]: Assuming this is intentional ie they both want to FOCUS_ANSWERS

File: frontend/src/utils/api/questions.js [hunk 5]
```diff
@@ -49,10 +49,12 @@ export function getAnswers(id) {
 
 /**
  * @param {String} id the id of the question being answered
- * @param {object} body the data associated with the new answer {title, body}
+ * @param {object} body the data associated with the new answer
  */
-export function postAnswer(id, body) {
-  return post(`/question/${id}`, { body })
+export function postAnswer(id, form) {
+  const ctypeHeaders = new Headers();
```
Comment (DarkPurple141) [inline frontend/src/utils/api/questions.js]: if you call this variable headers, you can write the line 'headers' shortly after. ie

```js
return post(`/question/${id}/answers`, {headers, body:JSON.stringify(form)})
```

Comment (DarkPurple141) [reply frontend/src/utils/api/questions.js]: Although might need to change the way we do requests so we don't interfere with any existing headers. (Something for another day).

File: frontend/src/store/questions/index.js [hunk 6]
```diff
@@ -43,9 +43,6 @@ const mutations = {
   API_ERROR (state, {code, message}) {
     state.error.code = code
     state.error.message = message
-  },
```
Comment (DarkPurple141) [inline frontend/src/store/questions/index.js]: I suppose this makes sense to always rely on server as source of truth, seems kinda odd to be rerendering the whole list on a single question change tho.

File: frontend/src/store/questions/constants.js [hunk 7]
```diff
@@ -1,23 +1,27 @@
 import {
   getQuestions,
   getQuestion,
-  getAnswers
+  getAnswers,
+  postAnswer
 } from '@/utils/api/questions'
 
 export const ACTIONS = {
```
Comment (DarkPurple141) [inline frontend/src/store/questions/constants.js]: Just a note on these symbols. They can (and probably should) be more descriptive as they're just unique key hashes but in debuggint they will give more relevant information. That was my bad for starting the convention.

File: frontend/public/index.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
Comment (DarkPurple141) [inline frontend/public/index.html]: Eventually populate with more meta tags.

File: backend/src/models/answer.js (hunk 2 above)
Comment (DarkPurple141) [reply backend/src/models/answer.js]: Sidenote I feel like calling things with exports.xxxx is sort of annoying.

See the way I made changes in db.js. Dunno!

File: frontend/src/store/questions/constants.js (hunk 4 above)
Comment (NunoDasNeves) [reply frontend/src/store/questions/constants.js]: yep

**Pull request merged**